infra/upload_images/container/handler.py

```
@@ -128,52 +127,218 @@ def _process_single(payload: Dict[str, Any]):
+                        logger.warning(
+                            f"Failed to create HTTP ChromaDB client: {http_error}. "
+                            f"Proceeding without ChromaDB fast-path."
+                        )
+
+        # Create embedding function for ChromaDB (if available)
+        if chroma_line_client:
+            def _embed_texts(texts):
+                if not texts:
+                    return []
+                if not os.environ.get("OPENAI_API_KEY"):
+                    raise RuntimeError("OPENAI_API_KEY is not set")
```

Ruff (0.14.4)

Abstract `raise` to an inner function

(TRY301)